lesson06/assignment/scripts/plantListForm.js

```diff
@@ -0,0 +1,153 @@
+  },
+  {genus: 'Myrica',
+    species: {
+      sciName: 'californica',
+      comName: 'Pacific wax myrtle',
+      src: './images/myrica_californica.jpg'}
+  },
+  {genus: 'Philadelphus',
+    species: {
+      sciName: 'lewisii',
+      comName: 'Mock Orange',
+      src: './images/philadelphus_lewisii.jpg'}
+  },
+  {genus: 'Ceanothus',
+    species: [
+      {sciName: 'arboreus', comName: 'California lilac',
+      src: './images/ceanothus_arboreus.jpg'},
+      {sciName: 'sanguineus', comName: 'Oregon tea tree',
+      src: './images/ceanothus_sanguineus.jpg'},
+      {sciName: 'thysiflorus', comName: 'var. Skylark',
+      src: './images/ceanothus_thysiflorus_skylark.jpg'},
+      {sciName: 'x pallidus', comName: 'var. Marie Simon',
+      src: './images/ceanothus_x_pallidus_Marie_Simon.jpg'}]
+    }];
+
+// Create HTML list with checkboxes
+availablePlants.forEach(function (e) {
+  const listGenus = document.createElement('div');
+    listGenus.setAttribute('class', 'accordion');
+    listGenus.setAttribute('id', 'accordion' + e.genus);
+
+  //document.querySelector('#available-plants').appendChild(listGenus);
+
+  // Next level div
+    const itemContainer = document.createElement('div');
+      itemContainer.setAttribute('class', 'accordion-item');
+
+    listGenus.appendChild(itemContainer);
+
+    // Accordion header
+      const itemHeader = document.createElement('h2');
+        itemHeader.setAttribute('class', 'accordion-header');
+        itemHeader.setAttribute('id', 'heading' + e.genus);
+
+      itemContainer.appendChild(itemHeader);
+
+      // Header button
+      const genusButton = document.createElement('button');
+        genusButton.setAttribute('class', 'accordion-button');
+        genusButton.setAttribute('type', 'button');
+        genusButton.setAttribute('data-bs-toggle', 'collapse');
+        genusButton.setAttribute('data-bs-target', '#collapse' + e.genus);
+        genusButton.setAttribute('aria-controls', 'collapse' + e.genus);
+        genusButton.innerHTML = e.genus;
+
+      // Accordion container
+      const listSpecies = document.createElement('div');
+        listSpecies.setAttribute('class', 'accordion-collapse collapse');
+        listSpecies.setAttribute('id', 'collapse' + e.genus);
+        listSpecies.setAttribute('data-bs-parent', '#accordion' + e.genus);
+        listSpecies.setAttribute('aria-labelledby', 'heading' + e.genus);
+
+      itemHeader.appendChild(genusButton);
+      itemContainer.appendChild(listSpecies);
+
+      const accordionContainer = document.createElement('div');
+        accordionContainer.setAttribute('class', 'accordion-body');
+
+      listSpecies.appendChild(accordionContainer);
+
+      // Species list items, with checkboxes
+      // First create variables for species objects & prep for forEach
+      let plantList
+      if(Array.isArray(e.species)) {
+        plantList = e.species;
+        } else {
+        plantList = [e.species];
+      }
```

Comment:
Nice! An alternative way to do this is to wrap e.species in brackets, then call `flat`, so if it was an array before, it stays one, and if it wasn't, it becomes one. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat 

```
// If e.species was already an array, it becomes a nested array, then flattens to single level.
plantList = [e.species].flat()
```